cadc-rest/src/main/java/ca/nrc/cadc/rest/RestServlet.java

```diff
@@ -449,34 +463,51 @@ void setAuthenticateHeaders(Subject subject, SyncOutput out, NotAuthenticatedExc
             return;
         }
         
-        AuthMethod am = AuthenticationUtil.getAuthMethod(subject);
-        if (am == null || AuthMethod.ANON.equals(am)) {
+        if (subject != null && !subject.getPrincipals().isEmpty()) {
+            // Authenticated...
+            
+            log.debug("Setting " + AuthenticationUtil.VO_AUTHENTICATED_HEADER + " header");
+            Set<HttpPrincipal> useridPrincipals = subject.getPrincipals(HttpPrincipal.class);
+            if (!useridPrincipals.isEmpty()) {
+                HttpPrincipal userid = useridPrincipals.iterator().next();
+                out.addHeader(AuthenticationUtil.VO_AUTHENTICATED_HEADER, userid.getName());
+                return;
+            }
+            Set<Principal> allPrincipals = subject.getPrincipals();
+            if (!allPrincipals.isEmpty()) {
+                Principal principal = allPrincipals.iterator().next();
```

Comment:
First, you don't need to check isEmpty() again here and could go straight to 
`subject.getPrincipals().iterator().next();`

Seems like this will ~randomly pick an identity type when HttpPrincipal aka userID is not known... practically that's going to be an OpenIdPrincipal, an X500Principal, or maybe a PosixPrincipal ... It could even be a CookiePrincipal or AuthorizationTokenPrincipal if the AuthentricatorImpl.validate(Subject) didn't do anything. Mostly cosmetic, but maybe put a comment like `// TODO: pick the most informative principal`